src/methods/call.js

```diff
@@ -316,72 +317,75 @@ export function OrdinaryCallEvaluateBody(
         throw new FatalError(err);
       }
     }
-  } else if (F.$FunctionKind === "generator") {
-    // 1. Perform ? FunctionDeclarationInstantiation(functionObject, argumentsList).
-    FunctionDeclarationInstantiation(realm, F, argumentsList);
-
-    // 2. Let G be ? OrdinaryCreateFromConstructor(functionObject, "%GeneratorPrototype%", « [[GeneratorState]], [[GeneratorContext]] »).
-    let G = OrdinaryCreateFromConstructor(realm, F, "GeneratorPrototype", {
-      $GeneratorState: undefined,
-      $GeneratorContext: undefined,
-    });
+  } else {
+    invariant(F instanceof ECMAScriptFunctionValue);
```

Comment:
Fyi, even using the OrdinaryFunctionValue as the signature, we still have to use invariant() here. Seems like flow is not smart enough to figure out branch typing.